### PR TITLE
Fix PositionComponent.toRect()

### LINKED ISSUE #970: flipping components - underlying rect on wrong position
# Current bug behaviour

The flipped component has its .toRect() not on the visual sprite but to the right of it. This bug manifests in many instances and causes weird behaviors of components jumping around and/or touch inputs not being passed correctly.

# Expected behaviour

component.toRect() should be at the same location as the visual sprite.

# Steps to reproduce

use a SpriteComponent and give it anchor.center
flip it with .flipHorizontally()

as expected the sprite component is flipped in the centerline but the underlying .toRect is not around the visual sprite but to the right of it. 

In one of my cases I for example used the mixin TapDetector and in the onTapDown method I used 

```
if (component
        .toRect()
        .contains(info.eventPosition.global.toOffset())) {...}
```

This condition becomes true if touched to the right of the visual sprite component on the screen. Its false if touched directly!

# More environment information

Flame version: 1.0.0-releasecandidate.14
Platform affected: android

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

Method `.toRect()` in `PositionComponent` now always returns the bounding rectangle of the component. Specifically, this method will now behave correctly even if the component was flipped, scaled, or rotated. When the component is rotated, we still return the smallest axis-aligned rectangle that is needed to circumscribe the component. This bounding rectangle will usually be larger in size than the dimensions of the component.

The method `.toAbsoluteRect()` is handled similarly.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, this is *not* a breaking change.

## Related Issues

Closes #970

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: packages/flame/lib/src/components/position_component.dart [hunk 1]
```diff
@@ -332,17 +332,45 @@ class PositionComponent extends Component {
...
+      final xs = [topLeft.x, topRight.x, bottomLeft.x, bottomRight.x]..sort();
+      final ys = [topLeft.y, topRight.y, bottomLeft.y, bottomRight.y]..sort();
+      return Rect.fromLTRB(xs.first, ys.first, xs.last, ys.last);
+    }
+  }
 
-  /// Mutates position and size using the provided [rect] as basis.
-  /// This is a relative rect, same definition that [toRect] use
-  /// (therefore both methods are compatible, i.e. setByRect ∘ toRect = identity).
+  // FIXME
```
Comment (erickzanardo) [inline packages/flame/lib/src/components/position_component.dart:373]: What is missing here?

File: packages/flame/lib/src/components/position_component.dart [hunk 2]
```diff
@@ -332,17 +332,45 @@ class PositionComponent extends Component {
...
+  }
 
-  /// Returns the absolute position/size of this component.
-  /// Absolute because it takes any possible parent position into consideration.
-  Rect toAbsoluteRect() => absoluteTopLeftPosition.toPositionedRect(scaledSize);
+  /// The bounding rectangle of the component in global coordinate space.
+  ///
+  /// This is similar to [toRect()], except the rectangle is projected into the
+  /// outermost coordinate frame.
+  Rect toAbsoluteRect() {
```
Comment (erickzanardo) [inline packages/flame/lib/src/components/position_component.dart:359]: This method and the other one are super similar, we should abstract it to an internal method so we don't repeat ourselves

Comment (luanpotter) [reply packages/flame/lib/src/components/position_component.dart:359]: +1. the "find axis aligned bounding box" bit could even be a generic method that we provide

File: packages/flame/lib/src/components/position_component.dart (hunk 1 above)
Comment (st-pasha) [reply packages/flame/lib/src/components/position_component.dart:373]: 1. It is not the inverse of `toRect()`: what if the component is rotated and/or scaled?
2. Do we want to reset the angle here?
3. If the component is flipped, do we want to unflip it, or keep as-is?
4. Changing the `size` of the component is questionable: it does not affect the visible size (unless the component uses its `size` property to render itself, which is not recommended), so do we want to modify the `scale` instead?

Generally, I'm not sure what the purpose of this method is and how it is expected to be used, so don't know in which direction to take it.

File: packages/flame/lib/src/components/position_component.dart (hunk 2 above)
Comment (st-pasha) [reply packages/flame/lib/src/components/position_component.dart]: ok

File: packages/flame/lib/src/components/position_component.dart (hunk 1 above)
Comment (erickzanardo) [reply packages/flame/lib/src/components/position_component.dart:363]: @luanpotter can you answser @st-pasha here?

Either we should not replace the original docs of the method for a FIXME. We could

 - If this is indeed wrong, put a FIXME comment with why it is broken, but leave the original dartdoc on the method
 - Fix the implementation and update the dartdoc if needed

Comment (st-pasha) [reply packages/flame/lib/src/components/position_component.dart]: Ok, I restored the comment to how it was before. The concerns about its functionality could be posted as a new issue instead.

Comment (st-pasha) [reply packages/flame/lib/src/components/position_component.dart]: Filed as #978

**Review by spydon — APPROVED** [review]
Lgtm!

**Review by wolfenrain — APPROVED** [review]
LGTM

**Pull request merged**